solidity/random-beacon/contracts/RandomBeaconGovernance.sol

```diff
@@ -0,0 +1,841 @@
+    ) {
+        /* solhint-disable not-rely-on-time */
+        require(changeInitiatedTimestamp > 0, "Change not initiated");
+        require(
+            block.timestamp - changeInitiatedTimestamp >= delay,
+            "Governance delay has not elapsed"
+        );
+        _;
+        /* solhint-enable not-rely-on-time */
+    }
+
+    constructor(RandomBeacon _randomBeacon) {
+        randomBeacon = _randomBeacon;
+    }
+
+    /// @notice Gets the time remaining until the governable parameter update
+    ///         can be committed.
+    /// @param changeTimestamp Timestamp indicating the beginning of the change.
+    /// @param delay Governance delay.
+    /// @return Remaining time in seconds.
+    function getRemainingChangeTime(uint256 changeTimestamp, uint256 delay)
+        internal
+        view
+        returns (uint256)
+    {
+        require(changeTimestamp > 0, "Change not initiated");
+        /* solhint-disable-next-line not-rely-on-time */
+        uint256 elapsed = block.timestamp - changeTimestamp;
+        if (elapsed >= delay) {
+            return 0;
+        } else {
+            return delay - elapsed;
+        }
+    }
+
+    /// @notice Begins the relay request fee update process.
+    /// @dev Can be called only by the contract owner.
+    /// @param _newRelayRequestFee New relay request fee
+    function beginRelayRequestFeeUpdate(uint256 _newRelayRequestFee)
+        external
+        onlyOwner
+    {
+        /* solhint-disable not-rely-on-time */
+        newRelayRequestFee = _newRelayRequestFee;
+        relayRequestFeeChangeInitiated = block.timestamp;
+        emit RelayRequestFeeUpdateStarted(_newRelayRequestFee, block.timestamp);
+        /* solhint-enable not-rely-on-time */
+    }
+
+    /// @notice Finalizes the relay request fee update process.
+    /// @dev Can be called only by the contract owner, after the governance
+    ///      delay elapses.
+    function finalizeRelayRequestFeeUpdate()
+        external
+        onlyOwner
+        onlyAfterGovernanceDelay(relayRequestFeeChangeInitiated, 24 hours)
+    {
+        randomBeacon.updateRelayEntryParameters(
+            newRelayRequestFee,
+            randomBeacon.relayEntrySubmissionEligibilityDelay(),
+            randomBeacon.relayEntryHardTimeout(),
+            randomBeacon.callbackGasLimit()
+        );
+        emit RelayRequestFeeUpdated(newRelayRequestFee);
+        relayRequestFeeChangeInitiated = 0;
+        newRelayRequestFee = 0;
+    }
+
+    /// @notice Begins the relay entry submission eligibility delay update
+    ///         process.
+    /// @dev Can be called only by the contract owner.
+    /// @param _newRelayEntrySubmissionEligibilityDelay New relay entry
+    ///        submission eligibility delay in blocks
+    function beginRelayEntrySubmissionEligibilityDelayUpdate(
+        uint256 _newRelayEntrySubmissionEligibilityDelay
+    ) external onlyOwner {
+        /* solhint-disable not-rely-on-time */
```

Comment:
We could consider silencing this rule in solhint, it is annoying.

Reply:
I agree, but I think we should wait till we have `solhint` enabled in `solidity/random-beacon`.
I added these  `/* solhint-disable not-rely-on-time */`statements because I thought they would be necessary.

Reply:
Agreed, I should be more clear - it's not something we should work on in this PR. I even think we should consider ignoring it in the Thesis-level solhint config.